src/main/java/org/veupathdb/service/eda/ds/plugin/standalonemap/StandaloneMapMarkersPlugin.java

```diff
@@ -112,94 +104,55 @@ void addRow(double lat, double lon, String overlayValue) {
+        overlayValuesProportion.put(overlayValue, proportion);
       }  
     }
 
-    String getOverlayValues(String valueSpec) {
-      List<BinRangeWithValue> overlayValuesResponse = new List<BinRangeWithValueImpl>();
+    List<BinRangeWithValue> getOverlayValues(String valueSpec) {
+      List<BinRangeWithValue> overlayValuesResponse = new ArrayList<>();
 
       for (String overlayValue : overlayValuesCount.keySet()) {
-        BinRangeWithValue newValue= new BinRangeWithValueImpl();
+        BinRangeWithValue newValue = new BinRangeWithValueImpl();
         newValue.setBinLabel(overlayValue);
-        newValue.setValue(valueSpec.equals("count") ? overlayValuesCount.get(overlayValue) : overlayValuesProportion.get(overlayValue));
+        newValue.setValue(valueSpec.equals("count") ? overlayValuesCount.get(overlayValue).get() : overlayValuesProportion.get(overlayValue));
         overlayValuesResponse.add(newValue);
       }
 
       return(overlayValuesResponse);
     }
   }
 
-  // TODO maybe this shoud live somewhere else? idk how generally useful it is
-  protected void validateBinRanges(List<BinRange> binRanges) throws ValidationException {
-    // TODO make sure we know how to compare bin start and end. this assumes they are numeric.
-    // BUT they probably arent numeric. they might be dates, which means they come in as strings
-    // I dont currently know how to differentiate numeric vs date bin ranges here
-    // either way they need type conversion...
-
-    // maybe there is a better way to do this type of thing??
-    boolean anyHaveBinStart = binRanges.stream().filter(bin -> bin.getBinStart() != null).findFirst().isPresent();
-    boolean anyHaveBinEnd = binRanges.stream().filter(bin -> bin.getBinEnd() != null).findFirst().isPresent();
-    boolean allHaveBinStart = !binRanges.stream().filter(bin -> bin.getBinStart() == null).findFirst().isPresent();
-    boolean allHaveBinEnd = !binRanges.stream().filter(bin -> bin.getBinEnd() == null).findFirst().isPresent();
-    if ((anyHaveBinStart | anyHaveBinEnd) & !(allHaveBinStart & allHaveBinEnd)) {
-      throw new ValidationException("All BinRanges must have binStart and binEnd if any BinRange has either.");
+  protected void validateOverlayConfig(OverlayConfig overlayConfig) throws ValidationException {
+    switch (overlayConfig.getOverlayType()) {
+      case CONTINOUS -> validateContinousBinRanges((ContinousOverlayConfig) overlayConfig);
+      case CATEGORICAL -> validateCategoricalOverlayValues((CategoricalOverlayConfig) overlayConfig);
     }
+  }
 
-    // start is always before end
-    boolean anyStartAfterEnd = binRanges.stream().filter(bin -> bin.getBinEnd() <= bin.getBinStart()).findFirst().isPresent();
-    if (anyHaveBinStart & anyStartAfterEnd) {
-      throw new ValidationException("All BinRanges must have binStart less than binEnd.");
+  private void validateCategoricalOverlayValues(CategoricalOverlayConfig overlayConfig) throws ValidationException {
```

Comment:
this answers a question i had somewhere else then.. except it might be okay to pass some cont vars as a categorical type overlay. itd be odd i guess, but not invalid, to say choose your 7 favorite values from some int w 12 unique values.